core/trino-spi/src/main/java/io/trino/spi/connector/MetadataProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import java.util.Optional;
+
+public interface MetadataProvider
+{
+    MetadataProvider NOOP_METADATA_PROVIDER = (session, tableName) -> Optional.empty();
+
+    /**
+     * Return relation schema or empty if relation does not exist.
```

Comment:
nit: javadoc kinda breaks abstraction. Assumin that CRM is just relation schema.

Reply:
changed to "relation metadata"